rules/cross-platform/privilege_escalation_setuid_setgid_bit_set_via_chmod.toml

```diff
@@ -31,59 +31,45 @@ tags = [
     "Data Source: Elastic Defend",
 ]
 timestamp_override = "event.ingested"
-type = "query"
-
+type = "eql"
 query = '''
-event.category:process AND event.type:(start OR process_started) AND
- process.name:chmod AND process.args:("+s" OR "u+s" OR /4[0-9]{3}/ OR g+s OR /2[0-9]{3}/) AND
- NOT process.args:
-           (
-             /.*\/Applications\/VirtualBox.app\/.+/ OR
-             /\/usr\/local\/lib\/python.+/ OR
-             /\/var\/folders\/.+\/FP.*nstallHelper/ OR
-             /\/Library\/Filesystems\/.+/ OR
-             /\/usr\/lib\/virtualbox\/.+/ OR
-             /\/Library\/Application.*/ OR
-             "/run/postgresql" OR
-             "/var/crash" OR
-             "/var/run/postgresql" OR
-             /\/usr\/bin\/.+/ OR /\/usr\/local\/share\/.+/ OR
-             /\/Applications\/.+/ OR /\/usr\/libexec\/.+/ OR
-             "/var/metrics" OR /\/var\/lib\/dpkg\/.+/ OR
-             /\/run\/log\/journal\/.*/ OR
-             \/Users\/*\/.minikube\/bin\/docker-machine-driver-hyperkit
-           ) AND
- NOT process.parent.executable:
-           (
-             /\/var\/lib\/docker\/.+/ OR
-             "/System/Library/PrivateFrameworks/PackageKit.framework/Versions/A/XPCServices/package_script_service.xpc/Contents/MacOS/package_script_service" OR
-             "/var/lib/dpkg/info/whoopsie.postinst"
-           )
+process where host.os.type == "linux" and event.type == "start" and event.action == "exec" and (
+  (process.name == "chmod" and (process.args : ("+s", "u+s", "g+s") or process.args regex "[24][0-9]{3}")) or
+  (process.name == "install" and process.args : "-m" and
+  (process.args : ("+s", "u+s", "g+s") or process.args regex "[24][0-9]{3}"))
+) and not (
+  process.parent.executable : (
+    "/usr/NX/*", "/var/lib/docker/*", "/var/lib/dpkg/info*", "/tmp/newroot/*",
+    "/System/Library/PrivateFrameworks/PackageKit.framework/Versions/A/XPCServices/package_script_service.xpc/Contents/MacOS/package_script_service"
```

Comment:
Odd to see it referencing a macOS package?

Reply:
It is cross-platform!